### PR TITLE
fix: add error check to fix the problem that the report is not sent to api

### PR DESCRIPTION
**Pull request by ardasnturk**
- updated `mutateAsync` structure to `onError` and `onSuccess` to better analyse error reports from api
- added `status` check to `onSuccess` section because most of the errors in api are not displayed in `onError` section
- added the `termsOfService` control we created inside `onSubmit` to the `alertTextGeneratorForMissingData` function to make the code more understandable
- added control of values in `postRequests` to not send unfilled content in the form to api

SUE-63

### CODE REVIEW

**Review by d4v3d4vE — APPROVED** [review]

File: src/queries/SUE/requests.tsx [hunk 1]
```diff
@@ -24,19 +24,20 @@ export const requests = async (queryVariables) => {
...
-  formData.append('long', data?.long);
-  formData.append('phone', data?.phone);
-  formData.append('service_code', data.serviceCode);
-  formData.append('title', data?.title);
+  data?.lastName && formData.append('last_name', data.lastName);
+  data?.lat && formData.append('lat', data.lat);
+  data?.long && formData.append('long', data.long);
+  data?.phone && formData.append('phone', data.phone);
+  data?.serviceCode && formData.append('service_code', data.serviceCode);
+  data?.title && formData.append('title', data.title);
```
Comment (donni106) [inline src/queries/SUE/requests.tsx:40]: can `data` be undefined here? isn't there always a `data` object passed in this method?

Comment (ardasnturk) [reply src/queries/SUE/requests.tsx:40]: In our previous control, we added a control against data being undefined. For example:
```
formData.append('title', data?.title);
```
 I used this control in the same way. After the changes made in the API, it started to not accept undefined data. So I decided to remove it from the form if there is no data.

Comment (donni106) [reply src/queries/SUE/requests.tsx:40]: ok, this way each line checks data. if you want to ensure that data is present what about a `if (data)` block?

Comment (ardasnturk) [reply src/queries/SUE/requests.tsx:40]: Yes, this would also make sense and would greatly reduce system requirements with a single control. 👍

Comment (donni106) [reply src/queries/SUE/requests.tsx:40]: @d4v3d4vE can you take over on this refactoring please?

Comment (donni106) [reply src/queries/SUE/requests.tsx:40]: done with a new commit, that will be cherry picked in master soon.

**Pull request merged**